internal/pkg/runtime/platform/metal/metal.go

```diff
@@ -38,6 +40,34 @@ func (m *Metal) Configuration() ([]byte, error) {
 		return nil, fmt.Errorf("no config option was found")
 	}
 
+	u, err := url.Parse(*option)
+	if err != nil {
+		return nil, err
+	}
+
+	values := u.Query()
+
+	if len(values) > 0 {
+		for key := range values {
+			switch key {
+			case "uuid":
+				s, err := smbios.New()
+				if err != nil {
+					return nil, err
+				}
+
+				uuid, err := s.SystemInformation().UUID()
```

Comment:
This value is not guaranteed to be filled, so I would caution against using it directly.

Reply:
Yeah this is something we have seen. The good news is that this is completely optional, and really only here for our metadata server we are building (for now). It should work _most_ of the time, but should be enough for now to get started on some of the bare metal stuff we are doing.